lib/ansible/executor/powershell/async_wrapper.ps1

```diff
@@ -145,11 +145,15 @@ try {
     $result_json = ConvertTo-Json -InputObject $result -Depth 99 -Compress
     Set-Content $results_path -Value $result_json
 
-    Write-AnsibleLog "INFO - waiting for async process to connect to named pipe for 5 seconds" "async_wrapper"
+    $np_timeout = $Payload.win_async_startup_timeout * 1000
+    Write-AnsibleLog "INFO - waiting for async process to connect to named pipe for $np_timeout milliseconds" "async_wrapper"
     $wait_async = $pipe.BeginWaitForConnection($null, $null)
-    $wait_async.AsyncWaitHandle.WaitOne(5000) > $null
+    $wait_async.AsyncWaitHandle.WaitOne($np_timeout) > $null
     if (-not $wait_async.IsCompleted) {
-        throw "timeout while waiting for child process to connect to named pipe"
+        $msg = "Ansible encountered a timeout while waiting for the async task to start and connect to the named"
```

Comment:
You still need a space at the end of this line or the beginning of the next, same with the next split.